ludwig/trainers/trainer.py

```diff
@@ -948,6 +948,8 @@ def _train_loop(
                     f"{psutil.Process(os.getpid()).memory_info()[0] / 1e6:0.2f}MB"
                 )
 
+            self.callback(lambda c: c.on_batch_end(self, progress_tracker, save_path))
```

Comment:
nit: Consider adding a note:

The `on_batch_end` callback happens before `run_evaluation` to get more accurate batch throughput measurements which exclude the evaluation step.

Reply:
Good call– done 👍